### PR TITLE
[1] Copy gateway server CLI to mlflow deployments start-server

### PR DESCRIPTION
**Pull request by dbczumar**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/dbczumar/mlflow/pull/10426?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10426/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10426
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Move gateway server CLI to mlflow deployments start-server

### How is this PR tested?

- [X] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [X] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/deployments/test_cli.py [hunk 1]
```diff
@@ -151,3 +151,50 @@ def test_explain_with_no_target_implementation(tmp_path):
...
+    name: invalid
+"""
+    )
+    res = runner.invoke(
+        cli.start_server,
+        ["--config-path", config],
+        catch_exceptions=False,
+    )
+    assert res.exit_code == 2
+    assert "The gateway configuration is invalid" in res.output
```
Comment (dbczumar) [inline tests/deployments/test_cli.py:199]: Renaming of error messages, etc. will be done in follow-up PRs

**Comment by dbczumar** [comment]
@mlflow-automation autoformat

**Review by harupy — APPROVED** [review]
LGTM

File: .github/workflows/deployments.yml [hunk 2]
```diff
@@ -33,4 +33,4 @@ jobs:
             pytest pytest-timeout pytest-asyncio httpx psutil
       - name: Run tests
         run: |
-          pytest tests/deploymnets/databricks
```
Comment (dbczumar) [inline .github/workflows/deployments.yml:36]: typo

File: .github/workflows/deployments.yml [hunk 3]
```diff
@@ -29,8 +29,9 @@ jobs:
       - uses: ./.github/actions/setup-python
       - name: Install dependencies
         run: |
+          pip install --no-dependencies tests/resources/mlflow-test-plugin && \
```
Comment (harupy) [inline .github/workflows/deployments.yml:32]:
```suggestion
          pip install --no-dependencies tests/resources/mlflow-test-plugin
```

File: tests/deployments/test_server_cli.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import pytest
+from click.testing import CliRunner
+
+from mlflow.deployments import cli
+
+pytest.importorskip("mlflow.gateway")
+runner = CliRunner()
```
Comment (harupy) [inline tests/deployments/test_server_cli.py:7]: Is this used?

Comment (dbczumar) [reply tests/deployments/test_server_cli.py:7]: looks like it isn't - removing

Comment (dbczumar) [reply tests/deployments/test_server_cli.py:7]:
```suggestion
```

File: mlflow/deployments/cli.py [hunk 5]
```diff
@@ -451,3 +453,46 @@ def get_endpoint(target, endpoint):
     for key, val in desc.items():
         click.echo(f"{key}: {val}")
     click.echo("\n")
+
+
+try:
+    from mlflow.gateway.config import _validate_config
```
Comment (harupy) [inline mlflow/deployments/cli.py:459]: Can we import this lazily in `validate_config_path`?

Comment (dbczumar) [reply mlflow/deployments/cli.py:459]: Yeah, I think that's better. Thanks - done!

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**